tests/jobs/localisation/example_case/scripts/common_functions.py

```diff
@@ -0,0 +1,548 @@
+            raise KeyError(f"Missing main keyword {main_key} in file: {yml_file_name}")
+        settings_yml_dict = spec_dict[main_key]
+        valid_keys1 = default_settings.keys()
+        for key1 in settings_yml_dict.keys():
+            if key1 in valid_keys1:
+                valid_keys2 = default_settings[key1].keys()
+                for key2 in settings_yml_dict[key1].keys():
+                    if key2 in valid_keys2:
+                        print(
+                            f"Modifying default setting for keyword '{key2}' "
+                            f"under '{key1}' in file {yml_file_name}  "
+                        )
+                        settings[key1][key2] = settings_yml_dict[key1][key2]
+                    else:
+                        raise KeyError(
+                            f"Unknown keyword '{key2}' specified under keyword '{key1}'"
+                            f" in {yml_file_name}"
+                        )
+            else:
+                raise KeyError(
+                    f"Unknown keyword '{key1}' specified under keyword '{main_key}'"
+                    f" in {yml_file_name}"
+                )
+    return settings
+
+
+def read_test_config(config_file_name):
+    if config_file_name is None:
+        spec_dict = None
+    else:
+        words = config_file_name.split(".")
+        if words[-1].upper() not in ["YML", "YAML"]:
+            raise IOError(
+                f"Expecting input yaml file as last argument for script {__file__}. "
+                f"Got the file {config_file_name}  "
+            )
+        with open(config_file_name, "r", encoding="utf-8") as yml_file:
+            spec_dict = yaml.safe_load(yml_file)
+
+    return specify_settings(spec_dict, config_file_name)
+
+
+def generate_field_and_upscale(settings, real_number):
+    seed_file_name = settings["field"]["seed_file"]
+    relative_std = settings["field"]["trend_relstd"]
+    use_trend = settings["field"]["trend_use"]
+    algorithm_method = settings["field"]["algorithm"]
+    start_seed = get_seed(seed_file_name, real_number)
+    if algorithm_method == "gstools":
+        print(f"Use algorithm: {algorithm_method}")
+        residual_field = simulate_field_using_gstools(settings, start_seed)
+    else:
+        print("Use algorithm: gaussianfft")
+        residual_field = simulate_field(settings, start_seed)
+    if use_trend == 1:
+        trend_field = trend(settings)
+        field3D = trend_field + relative_std * residual_field
+
+    else:
+        field3D = residual_field
+
+    # Write field parameter for fine scale grid
+    field_object = export_field(settings, field3D)
+    field_values = field_object.values
+
+    # Calculate upscaled values for selected coarse grid cells
+    upscaled_values = upscaling(
+        field_values,
+        settings,
+        write_field=True,
+        iteration=0,
+    )
+    return upscaled_values
+
+
+def get_seed(seed_file_name, r_number):
```

Comment:
The #noqa: E722 directive is used to tell linters like flake8 to ignore the bare except clause, which is not a recommended practice in Python.

To be more specific in your exception handling, catch only the exceptions you expect, in this case, the IndexError when trying to access an element from the lines list that doesn't exist and the ValueError when trying to convert a non-integer string to an integer.

```python
def get_seed(seed_file_name, r_number):
    with open(seed_file_name, "r", encoding="utf8") as file:
        lines = file.readlines()
        try:
            seed_value = int(lines[r_number - 1])
        except IndexError:
            raise IOError(f"Seed value not found for realization {r_number}")
        except ValueError:
            raise IOError(f"Invalid seed value in file for realization {r_number}")
    return seed_value
```

Reply:
OK